### PR TITLE
Feat/error handling execution

### LINKED ISSUE #138: feat: Only the current execution context should be reverted if not enough values are on the Stack
# Feature Request

**Describe the Feature Request**
Most opcodes modify the state of the current execution context. One important thing is that if there are not enough values on the Stack when executing an opcode (e.g. there is only one value on the Stack and you try to execute ADD), then only the current execution context should be reverted, but the whole tx should not be reverted (as per https://www.evm.codes/ indications).

Currently, using `stack.pop_n()` will revert the entire tx if there are not enough values on the stack.

**Describe Preferred Solution**
TBD

**Additional Context**

**If the feature request is approved, would you be willing to submit a PR?**
_(Help can be provided if you need assistance submitting a PR)_

- [x] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by Eikix**
Light PR to start thinking about result based error management in execution context.

## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves #138 

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

## Added changes to [`CHANGELOG.md`](../CHANGELOG.md)

- [x] Yes
- [ ] No

### CODE REVIEW

File: src/tests/test_kakarot.cairo [hunk 1]
```diff
@@ -5,20 +5,3 @@ use kakarot::evm;
 use kakarot::stack::StackTrait;
 use kakarot::context::CallContext;
 use kakarot::errors;
-
-
```
Comment (Eikix) [inline src/tests/test_kakarot.cairo:9]: this test is no longer relevant as execute does not panic anymore, 
and evm::execute does not return an error yet

File: src/interpreter.cairo [hunk 2]
```diff
@@ -42,132 +45,149 @@ impl EVMInterpreterImpl of EVMInterpreterTrait {
...
+                if context.reverted() { // TODO: Revert logic
+                }
+                if context.stopped() { // TODO: stopped logic
+                }
+            },
+            Result::Err(error) => {
+                // If an error occurred, revert execution context.
+                // Currently, revert reason is a Span<u8>. 
+                context.revert(u256_to_bytes_array(error.into()).span());
+                self.run(ref context);
```
Comment (enitrat) [inline src/interpreter.cairo:66]: do we run the next instruction after a revert? Shouldn't we handle revert instead

Comment (Eikix) [reply src/interpreter.cairo:66]: this is to not duplicate code with an revert that may be triggered by REVERT opcode

Comment (enitrat) [reply src/interpreter.cairo:66]: But I don't get it, normally you should revert straight away and not continue the execution any further

Comment (Eikix) [reply src/interpreter.cairo:66]: Fair point

Comment (Eikix) [reply src/interpreter.cairo:66]: ah yeah, my bad though it doenst work here

**Review by enitrat — APPROVED** [review]

**Pull request merged**